snapshot_creator.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2021 the v8go contributors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package v8go
+
+// #include <stdlib.h>
+// #include "v8go.h"
+import "C"
+import "unsafe"
+
+type FunctionCodeHandling int
+
+const (
+	FunctionCodeHandlingKlear FunctionCodeHandling = iota
+	FunctionCodeHandlingKeep
+)
+
+type StartupData struct {
+	ptr *C.SnapshotBlob
+}
```

Comment:
The user of the library will need to be able to both get a byte array for serialization and to be able to construct the type for passing back for creating an isolate from it.  So perhaps this should just be a byte slice.

```suggestion
type StartupData []byte
```

that would make it simpler to use, since there would be no need to garbage collect the data.  When creating the isolate from this data, we would just need to store a reference to the byte slice in the isolate so that it isn't garbage collected while it is in use (assuming it needs to outlive the isolate it was used to initialize)